x-pack/plugins/cases/server/connectors/cases/cases_connector.ts

```diff
@@ -108,15 +109,15 @@ export class CasesConnector extends SubActionConnector<
     /**
      * Add circuit breakers to the number of oracles they can be created or retrieved
      */
-    const oracleRecords = await this.upsertOracleRecords(
+    const oracleRecordsMap = await this.upsertOracleRecords(groupedAlertsWithOracleKey);
+    const oracleRecordMapWithTimeWindowHandled = await this.handleTimeWindow(
```

Comment:
@adcoelho As we discussed I moved the logic of the time window outside the upsert method.